imports/client/TestRunner.js

```diff
@@ -0,0 +1,190 @@
+            ).show();
+        }
+    }
+}
+
+class TestCase {
+    static testCases = [];
+
+    constructor(testName, triggers, condition, test) {
+        this.testName = testName; //name of the test for list generation
+        this.triggers = triggers; //to determine to which scenarios applies
+        this.condition = condition; //checks if the test is to be run at all. For future purposes e.g. disabling tests via settings
+        this.test = test; // the test itself. Receives the minute object as a parameter, returns a string if the test fails, otherwise undefined
+    }
+
+    static createTestCases() {
+        // to add tests simply push a new TestCase Object to the testCases property
+
+        // no topics in minute (F) (A)
+        TestCase.testCases.push(new TestCase('No topics in minute',
+            [TestRunner.TRIGGERS.finalize, TestRunner.TRIGGERS.sendAgenda],
+            () => {return true;},
+            (minute) => {
+                if (!minute.topics || minute.topics.length === 0)
+                    return 'This minute has no topics';
+            }
+        ));
+
+        // no participant marked as present (F)
+        TestCase.testCases.push(new TestCase('No participants marked as present',
+            TestRunner.TRIGGERS.finalize,
+            () => {return true;},
+            (minute) => {
+                let noParticipantsPresent = true;
+                minute.participants.forEach(p => {
+                    if(p.present) noParticipantsPresent = false;
+                    }
+                );
+                if(noParticipantsPresent)
+                    return 'No participant is marked as present'
+            }
+        ));
+
+        // no topics checked (F)
+        TestCase.testCases.push(new TestCase('No topic is checked',
+            TestRunner.TRIGGERS.finalize,
+            () => {return true;},
+            (minute) => {
+                if(minute.topics.length < 0) return;
+                let noTopicChecked = true;
+                minute.topics.forEach(topic => {
+                        if(!topic.isOpen) noTopicChecked = false;
+                    }
+                );
+                if(noTopicChecked)
+                    return 'No topic is checked'
+            }
+        ));
+
+        // topic checked but no children (F)
+        TestCase.testCases.push(new TestCase('A topic is checked but has no children',
+            TestRunner.TRIGGERS.finalize,
+            () => {return true;},
+            (minute) => {
+                if(minute.topics.length < 0) return;
+
+                let checkedButChildren = false;
+                minute.topics.forEach(topic => {
+                        if(topic.isOpen) return;
+                        if(!topic.infoItems || topic.infoItems.length === 0) checkedButChildren = true;
+                    }
+                );
+                if(checkedButChildren)
+                    return 'A topic is checked but has no children'
```

Comment:
If the meeting has >30 topics (which may happen), it is unclear which topic has problems. Could you add hints where you show the first e.g., 20 chars of the topic title. "Discuss team event for ..." - and truncate with elippsis. Make the number 20 configurable constant.